FileGetContentsLoader.php

```diff
@@ -27,95 +27,134 @@ public function loadDocument($url)
+                            return $this->retrieveRemoteDocument($link['uri']);
+                            break;
                     }
                 }
             }
+        }
 
-            $linkHeaderValues = $this->parseContextLinkHeaders($linkHeaderValues, new IRI($url));
+        $linkHeaderValues = $this->parseContextLinkHeaders($linkHeaderValues, new IRI($url));
 
-            if (count($linkHeaderValues) === 1) {
-                $remoteDocument->contextUrl = reset($linkHeaderValues);
-            } elseif (count($linkHeaderValues) > 1) {
-                throw new JsonLdException(
-                    JsonLdException::MULTIPLE_CONTEXT_LINK_HEADERS,
-                    'Found multiple contexts in HTTP Link headers',
-                    $http_response_header
-                );
-            }
-
-            // If we got a media type, we verify it
-            if ($remoteDocument->mediaType) {
-                // Drop any media type parameters such as profiles
-                if (false !== ($pos = strpos($remoteDocument->mediaType, ';'))) {
-                    $remoteDocument->mediaType = substr($remoteDocument->mediaType, 0, $pos);
-                }
+        if (count($linkHeaderValues) === 1) {
+            $remoteDocument->contextUrl = reset($linkHeaderValues);
+        } elseif (count($linkHeaderValues) > 1) {
+            throw new JsonLdException(
+                JsonLdException::MULTIPLE_CONTEXT_LINK_HEADERS,
+                'Found multiple contexts in HTTP Link headers',
+                $http_response_header
+            );
+        }
 
-                $remoteDocument->mediaType = trim($remoteDocument->mediaType);
-
-                if ('application/ld+json' === $remoteDocument->mediaType) {
-                    $remoteDocument->contextUrl = null;
-                } elseif (('application/json' !== $remoteDocument->mediaType) &&
-                    (0 !== substr_compare($remoteDocument->mediaType, '+json', -5))) {
-                    throw new JsonLdException(
-                        JsonLdException::LOADING_DOCUMENT_FAILED,
-                        'Invalid media type',
-                        $remoteDocument->mediaType
-                    );
-                }
+        // If we got a media type, we verify it
+        if ($remoteDocument->mediaType) {
+            // Drop any media type parameters such as profiles
+            if (false !== ($pos = strpos($remoteDocument->mediaType, ';'))) {
+                $remoteDocument->mediaType = substr($remoteDocument->mediaType, 0, $pos);
             }
 
-            $remoteDocument->document = Processor::parse($input);
+            $remoteDocument->mediaType = trim($remoteDocument->mediaType);
 
-            return $remoteDocument;
+            if ('application/ld+json' === $remoteDocument->mediaType) {
+                $remoteDocument->contextUrl = null;
+            } elseif ('application/octet-stream' === $remoteDocument->mediaType) {
```

Comment:
This is not ideal, but appears to be what schema.org actually returns...

Reply:
This was fixed:

https://github.com/schemaorg/schemaorg/issues/2595

Reply:
I still get the error saying it is getting text/html back.

Reply:
Oh I see maybe you were saying that specific part of the problem was fixed.